runtime/port/j9omrport/j9omrport.mk.ftl

```diff
@@ -1,48 +0,0 @@
-
-#
-# Copyright (c) 2012, 2017 IBM Corp. and others
-#
-# This program and the accompanying materials are made available under
-# the terms of the Eclipse Public License 2.0 which accompanies this
-# distribution and is available at https://www.eclipse.org/legal/epl-2.0/
-# or the Apache License, Version 2.0 which accompanies this distribution and
-# is available at https://www.apache.org/licenses/LICENSE-2.0.
-#
-# This Source Code may also be made available under the following
-# Secondary Licenses when the conditions for such availability set
-# forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
-# General Public License, version 2 with the GNU Classpath
-# Exception [1] and GNU General Public License, version 2 with the
-# OpenJDK Assembly Exception [2].
-#
-# [1] https://www.gnu.org/software/classpath/license.html
-# [2] http://openjdk.java.net/legal/assembly-exception.html
-#
-# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
-#
-
-# This makefile is generated using an UMA template.
-
-# UMA_PATH_TO_ROOT can be overridden by a command-line argument to 'make'.
-UMA_PATH_TO_ROOT ?= ../../
-OMR_DIR ?= $(UMA_PATH_TO_ROOT)omr
-
-# top_srcdir is the top of the omr source tree
-top_srcdir := $(OMR_DIR)
-include $(top_srcdir)/omrmakefiles/configure.mk
-
-# PORT_SRCDIR is the location of the OMR port source code.
-PORT_SRCDIR := $(top_srcdir)/port/
-
-MODULE_NAME := j9omrport
-ARTIFACT_TYPE := archive
-
-<#if uma.spec.flags.port_omrsigSupport.enabled>
-MODULE_CPPFLAGS += -DOMRPORT_OMRSIG_SUPPORT
-MODULE_INCLUDES += $(UMA_PATH_TO_ROOT)include
-MODULE_SHARED_LIBS += omrsig
```

Comment:
Where does this configuration end up with this change?  The OpenJ9 port library needs omrsig and the signal support.

Reply:
aah! didn't catch that. Thanks for pointing it out.

Let me see how this can be handled.

Just thinking aloud...one possibility is to remove OMRPORT_OMRSIG_SUPPORT flag as its always on for all specs. Now this would cause omrsig support to be enabled by default in OMR, which may not not be desired by every consumer of OMR.
But this can be handled at runtime as all function calls related to omrsig are protected by runtime flag OMRPORT_SIG_OPTIONS_OMRSIG_NO_CHAIN flag.
eg: https://github.com/eclipse/omr/blob/951864524077a4fbd26e2fe80adb9a2c8cc21ee6/port/unix/omrsignal.c#L703-L709

We can update OMR to enable this flag by default. Note that currently the flag is off by default.
Any consumer of OMR that wants omrsig support, like OpenJ9, can then disable it explicitly.

Does that make sense? Did I miss anything again?

Reply:
I would prefer not to enable this by default in OMR.

Reply:
> I would prefer not to enable this by default in OMR.

Did you mean not enabling `OMRPORT_SIG_OPTIONS_OMRSIG_NO_CHAIN` flag by default?
In that case, omrsig support will be enabled by default on OMR if we remove `OMRPORT_OMRSIG_SUPPORT` macro.
If that is acceptable, then just removing the macro `OMRPORT_OMRSIG_SUPPORT` along with the existing changes should be sufficient for linking with `libj9prtstatic.a` along with omrsig support.

Reply:
@charliegracie @DanHeidinga what do you think about removing the macro OMRPORT_OMRSIG_SUPPORT and have omrsig support enabled by default?